client/src/components/ResourceHub/AddResourceModal.tsx

```diff
@@ -0,0 +1,110 @@
+import React, { ReactElement, FC, Dispatch } from 'react';
+import { Modal, Button, Form, Input, Checkbox } from 'semantic-ui-react';
+
+import { createResource, isError } from '../../utils/apiWrapper';
+
+import '../../css/AddResourceModal.css';
+
+const teams = [
+  'General',
+  'Editing',
+  'Factchecking',
+  'Illustration',
+  'Photography',
+  'Onboarding',
+  'Visuals',
+  'Writing',
+];
+
+interface IProps {
+  onAdd: Dispatch<void>;
+}
+
+const AddResourceModal: FC<IProps> = ({ onAdd }): ReactElement => {
+  const [open, setOpen] = React.useState<boolean>(false);
+  const [resourceTitle, setResourceTitle] = React.useState<string>('');
+  const [resourceURL, setResourceURL] = React.useState<string>('');
+  const [selectedTags, setSelectedTags] = React.useState<Set<string>>(
+    new Set(),
+  );
+
+  const handleTagSelect = (tag: string): void => {
+    const newTags = new Set(selectedTags);
+    if (newTags.has(tag)) {
+      newTags.delete(tag);
+    } else {
+      newTags.add(tag);
+    }
+    setSelectedTags(newTags);
+  };
+
+  async function addResource(): Promise<void> {
```

Comment:
@alicesf2 traditional functions or  arrow functions? I figured we would be using arrow because functional react and realized I never asked.

`const addResource = async (): Promise<void> => {...}`